### ISSUE TITLE
v1.11.2: startDestroyBlock hook not called + 2 reasonable suggestions ^^

### ISSUE THREAD

**Issue by Sin0psysS** [zhuowei/MCPELauncher#487]
Hello zhuoweizhangangouiaeiwh! :D
As it says in the title, I'd like to report that in the play store version of blocklauncher (1.11.2), the startDestroyBlock hook doesn't seem to work anymore.
I just released [a mod using it a lot](http://www.minecraftforum.net/forums/minecraft-pocket-edition/mcpe-mods-tools/2593066-0-13-momentum-a-powerful-mod-menu-inventory) and then realised it didn't work, so it would be great if you fixed it! :)

There are also two things I think would be useful to add:
1) Correct me if I'm wrong, and I tried a lot of ways, but I don't think there's any way to detect if a chest is a large one or not :3
2) My mod sorts inventories and chests, by scanning the inventory and reading the id, data and amount of each item in the container and then arranging them.
But since there's no way for it to detect if the items are enchanted, those get lost in the process! :/
So maybe you could add a Player.getInventorySlotEnchantments/Level.getChestSlotEnchantments method, returning an array of the enchantments with their amplifier?
As well as a method to enchant items of a specifical slot, of course.

Thanks for reading me, stay awesome and have a nice day!

**Comment by Izaac9236** [comment]
Hmmmm good idea but, tge jonhOarilketa is noob lololol

**Comment by Sin0psysS** [comment]
What do you mean?

**Comment by Black-DeathPE** [comment]
On the chest tapped look north, east, south and west of it, to see if there's a chest there. This wouldn't work for chests next to each other, though, so your suggestion would work much better than this.

**Comment by Sin0psysS** [comment]
Yep I tried that and came to the same conclusion: I preferred not to do that because reading a non existing slot in a chest crashes the game, even in a try/catch

**Comment by Sin0psysS** [comment]
Edited the first post, enchantment methods are out, thank you :D
